### PR TITLE
Copy configuration files - Initial code

### PR DESCRIPTION
**Pull request by zeshanziya**
This is the initial code for copying configuration files. Please do not merge.

### CODE REVIEW

**Review by hussainweb — COMMENTED** [review]
My concern in using the `activate` method is that I recall reading some limitations in what we can reasonably do within `activate`. I don't know if there is a harm here though and I can't find that reference.

I understand we need to run before GrumPHP. I wonder if there is a way we can use a hook earlier than the `install` command. Or we use the same hook (and the `update` hook) but change the order of execution with some kind of priority. If this takes a lot of time, let's go with `activate` for now and we can create a follow-up issue.

File: src/Composer/AxelerantDQCPlugin.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+     */
+    public function copyFilesToProject(): void
+    {
+        // Determine the destination directory in the project
+        $destination = getcwd();
+
+        // Copy each file to the project
+        copy(__DIR__ . '/../../grumphp.yml.dist', $destination . '/grumphp.yml.dist');
+        copy(__DIR__ . '/../../phpcs.xml.dist', $destination . '/phpcs.xml.dist');
+        copy(__DIR__ . '/../../phpmd.xml.dist', $destination . '/phpmd.xml.dist');
```
Comment (hussainweb) [inline src/Composer/AxelerantDQCPlugin.php:113]: We need to figure out what to do when the file already exists. If we always overwrite, we might lose project level customizations.

The simplest approach might be that this plugin outputs warnings that the files already exist and we are not overwriting them. To get the latest changes, give them a `cp` command to get the latest file.

**Comment by zeshanziya** [comment]
> lot

I believe the priority is within the plugin itself for multiple methods for a single event. But I can give it a try if priority works across multiple composer plugin. If the priority method not working and there is no other way, then I will use the activate callback. I didn't go too deep on how to activate callback works, but I tried and the copying file was working.

File: src/Composer/AxelerantDQCPlugin.php (hunk 1 above)
Comment (hussainweb) [reply src/Composer/AxelerantDQCPlugin.php:103]: For now, let's go ahead with warn and overwrite. We can think more about this later.

Comment (zeshanziya) [reply src/Composer/AxelerantDQCPlugin.php:103]: Yes, we've included the 'cp' command in the customize section of the README, which renames files and removes the 'dist' extension. I suggest we should not add it to .gitignore. This allows developers to be informed of any changes, enabling them to incorporate modifications into customized files if any.

Ok, I will overwrite with a warning.

**Pull request closed without merging**